Automation update

mokagio opened the pull request:
### Fix
***(Required)*** Add a concise description of what you fixed.  If this is related to an issue, add a link to it.  If applicable, add screenshots, animations, or videos to help illustrate the fix.

### Test
***(Required)*** List the steps to test the behavior.  For example:
> 1. Go to...
> 2. Tap on...
> 3. See error...

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
> `RELEASE-NOTES.txt` was updated in d3adb3ef with:
> 
> > Added markdown support

If the changes should not be included in release notes, add a statement to this section. For example:

> These changes do not require release notes.

mokagio committed:
Use shared vars in Buildkite pipilines and explicitly set `mac` queue

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Use `set_milestone_frozen_marker` over `setfrozentag`

mokagio committed:
Use `xcconfig` to set `DEVELOPMENT_TEAM`

mokagio committed:
Remove unused `PROVISIONING_PROFILE` build setting

mokagio committed:
Use `xcconfig` to define provisioning profile for release builds

We'll thinking about dev builds later, at the moment, we're focused on
the release workflow

mokagio committed:
Use `xcconfig` to specify code sign identity and make explicit

mokagio committed:
Update Fastlane to latest version 2.222.0

mokagio committed:
Use full path for build folder

dangermattic commented on fastlane/lanes/build.rb:
```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+platform :mac do
+  lane :build_and_upload_app_store do |options|
```
<table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Automattic/simplenote-macos/blob/34ff326c579801ec45af5c8956c0014425743887/fastlane/lanes/build.rb#L4"></span>Lint/UnusedBlockArgument: Unused block argument - <code>options</code>. You can omit the argument if you don't care about it.</td>
    </tr>
  </tbody>
</table>

mokagio committed:
Enable hardened runtime for release, and move to `xcconfig`

Required for notarization in Xcode 15.4

mokagio committed:
Add lanes for getting Developer ID as well as ASC certificates
